gwdetchar/io/html.py

```diff
@@ -68,6 +81,79 @@ def write_param(param, value):
     return page()
 
 
+def render_code(code, language):
+    """Render a black of code with syntax highlighting
+
+    Parameters
+    ----------
+    code : `str`
+        a raw block of source code
+
+    language : `str`
+        language the code is written in, e.g. `'python'`
+    """
+    lexer = get_lexer_by_name(language, stripall=True)
+    return highlight(code, lexer, FORMATTER)
+
+
+def get_command_line(language='bash'):
+    """Render the command line invocation used to generate a page
+
+    Parameters
+    ----------
+    language : `str`, optional
+        language the code is written in, default: `'bash'`
+    """
+    commandline = ' '.join(sys.argv)
+    return render_code(commandline, language)
+
+
+def write_footer(about=None, date=None, class_=False,
+                 linkstyle='color:#000;'):
+    """Write a <footer> for a bootstrap page
+
+    Parameters
+    ----------
+    about : `str`, optional
+        path of about page to link
+
+    date : `datetime.datetime`, optional
+        the datetime representing when this analysis was generated, defaults
+        to `~datetime.datetime.now`
+
+    Returns
+    -------
+    page : `~MarkupPy.markup.page`
+        the markup object containing the footer HTML
+    """
+    page = markup.page()
+    if class_:
+        page.twotags.append('footer')
+        markup.element('footer', case=page.case, parent=page)(class_='footer')
+    page.div(class_='container')
```

Comment:
Related to above, `<div class="container">` shouldn't really be nested inside an existing `container` - this is what causes the indentation in the text, which (IMO) is unsightly.